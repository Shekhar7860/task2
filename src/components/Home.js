import { FlatList, Image, SafeAreaView, Text, TextInput, TouchableOpacity, View } from 'react-native'
import React,{ useEffect, useState } from 'react';

import { items } from '../data/data';

const Home = () => {
    const [search,setSearch]=useState(false)
    const [filteredDataSource, setFilteredDataSource] = useState(items);
    const [masterDataSource, setMasterDataSource] = useState(items);
    const searchFilterFunction = (text) => {
        // Check if searched text is not blank
        if (text) {
          // Inserted text is not blank
          // Filter the masterDataSource
          // Update FilteredDataSource
          const newData = masterDataSource.filter(
            function (item) {
              const itemData = item.name
                ? item.name.toUpperCase()
                : ''.toUpperCase();
              const textData = text.toUpperCase();
              return itemData.indexOf(textData) > -1;
          });
          console.log(newData)
          setFilteredDataSource(newData);
          setSearch(text);
        } else {
          // Inserted text is blank
          // Update FilteredDataSource with masterDataSource
          setFilteredDataSource(masterDataSource);
          setSearch(text);
        }
      };
    return(<SafeAreaView style={{backgroundColor:'#000000'}}>
        {!search ?<View style={{flexDirection:'row',marginTop:20,justifyContent:'space-between',width:'90%',alignItems:'center'}}>
            <Image style={{width:30,height:30,marginLeft:15}} source={require('../images/Back.png')}/>
            <Text style={{color:'#ffffff',fontSize:25}}>Romantic Comedy</Text>
            <TouchableOpacity onPress={() => setSearch(true)}>
            <Image style={{width:30,height:30}} source={require('../images/search.png')}/>
            </TouchableOpacity>
        </View>
        :<View style={{flexDirection:'row',marginTop:20,justifyContent:'space-between',width:'95%'}}>
            <TextInput    onChangeText={(text) => searchFilterFunction(text)} placeholder='Enter text' placeholderTextColor="#ffffff" style={{paddingLeft:15,marginLeft:5,width:'85%',borderWidth:1,borderColor:'#ffffff',color:'#ffffff'}}/>
        <TouchableOpacity onPress={() => setSearch(false)}>
            <Image style={{width:40,height:40,tintColor:'#ffffff',marginTop:5}} source={require('../images/close.png')}/>
            </TouchableOpacity>
        </View>}

        <View style={{margin:30,height:1000}}>
        <FlatList
        data={filteredDataSource}
        numColumns={3}
        extraData={filteredDataSource}
        keyExtractor={(item, index) => String(index)}
        renderItem={(item) =>  
        <View style={{justifyContent:'space-between',flex:1,marginBottom:60,marginTop:10}}>
        <Image style={{width:100,height:150}} source={item.item.posterimage}/>
        <Text style={{width:100,color:'#ffffff',marginTop:10}}>{item.item.name}</Text></View>}   
/>
</View></SafeAreaView>)
}




export default Home;