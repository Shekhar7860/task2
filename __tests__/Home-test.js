import 'react-native';

import Home from '../src/components/Home';
import React from 'react';
import renderer from 'react-test-renderer';

// Note: test renderer must be required after react-native.


it('renders correctly', () => {
  renderer.create(<Home />);
});
it('creates snapshot', () => {
    const tree =renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
  });