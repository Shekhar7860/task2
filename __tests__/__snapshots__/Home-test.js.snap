// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#000000",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/Back.png",
        }
      }
      style={
        Object {
          "height": 30,
          "marginLeft": 15,
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 25,
        }
      }
    >
      Romantic Comedy
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/images/search.png",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": 1000,
        "margin": 30,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster1.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster2.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster3.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster4.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster5.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster6.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster7.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster8.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster9.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster1.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster2.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster3.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster4.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster5.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster6.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster7.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster8.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster9.jpg",
            },
          },
        ]
      }
      extraData={
        Array [
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster1.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster2.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster3.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster4.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster5.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster6.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster7.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster8.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster9.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster1.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster2.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster3.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster4.jpg",
            },
          },
          Object {
            "name": "Family Pot",
            "posterimage": Object {
              "testUri": "../../../src/images/poster5.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster6.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster7.jpg",
            },
          },
          Object {
            "name": "Rear Window",
            "posterimage": Object {
              "testUri": "../../../src/images/poster8.jpg",
            },
          },
          Object {
            "name": "The Birds",
            "posterimage": Object {
              "testUri": "../../../src/images/poster9.jpg",
            },
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster1.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                The Birds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster2.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Rear Window
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster3.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Family Pot
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster4.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Family Pot
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster5.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Rear Window
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster6.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                The Birds
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster7.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Rear Window
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster8.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                The Birds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster9.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Family Pot
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster1.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                The Birds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster2.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                The Birds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster3.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Rear Window
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster4.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Family Pot
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster5.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Family Pot
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster6.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Rear Window
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster7.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                The Birds
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster8.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                Rear Window
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "space-between",
                  "marginBottom": 60,
                  "marginTop": 10,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/images/poster9.jpg",
                  }
                }
                style={
                  Object {
                    "height": 150,
                    "width": 100,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#ffffff",
                    "marginTop": 10,
                    "width": 100,
                  }
                }
              >
                The Birds
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
